### ISSUE TITLE
False positive: type struct is unused

### ISSUE THREAD

**Issue by leventov** [golangci/golangci-lint#1199]
foo.go:
```go
package main

import "log"

func main() {
	foo := &foo{}
	log.Println(foo)
}

type foo struct {}
```

```
$ golangci-lint --version
golangci-lint has version 1.27.0 built from fb74c2e on 2020-05-13T18:45:55Z
$ golangci-lint run foo.go
foo.go:5:6: `main` is unused (deadcode)
func main() {
     ^
foo.go:10:6: type `foo` is unused (unused)
```

Might be the same as #1193

**Comment by sayboras** [comment]
@leventov seems a little bit weird for me, I tried to replicate the above issue but can't

```bash script
$ cat foo.go 
package main

import "log"

func main() {
	foo := &foo{}
	log.Println(foo)
}

type foo struct {}

$ golangci-lint run foo.go
```

Just want to confirm if your package is _main_ or not. Please find below the snippet.

```bash script
$ cat foo.go 
package dummy 

import "log"

func main() {
	foo := &foo{}
	log.Println(foo)
}

type foo struct {}

$ golangci-lint run foo.go
foo.go:5:6: `main` is unused (deadcode)
func main() {
     ^
foo.go:10:6: type `foo` is unused (unused)
```
